Fix kafka internal docker connection (#2490)
testcontainers/testcontainers-go: docs/modules/kafka.md

{% include "../features/common_functional_options.md" %}

#### ClusterId

You can set up cluster id by using `WithClusterID` option.

```
KafkaContainer, err = kafka.RunContainer(ctx,
		kafka.WithClusterID("test-cluster"),
		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"))
```

#### Listeners

If you need to connect new listeners, you can use `WithListener(listeners []KafkaListener)`. 

> This option controls next env parameters: 

Reviewer: ```suggestion
This option controls the following environment variables for the Kafka container: 
```

- `KAFKA_LISTENERS`
- `KAFKA_REST_BOOTSTRAP_SERVERS`
- `KAFKA_LISTENER_SECURITY_PROTOCOL_MAP`
- `KAFKA_INTER_BROKER_LISTENER_NAME`
- `KAFKA_ADVERTISED_LISTENERS`

Example:
```
KafkaContainer, err = kafka.RunContainer(ctx,
		kafka.WithClusterID("test-cluster"),
		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
		network.WithNetwork([]string{"kafka"}, Network),
		kafka.WithListener([]kafka.KafkaListener{
			{
				Name: "INTERNAL",
				Ip:   "kafka",
				Port: "9092",
			},
		}),
	)

> ```

Reviewer: Let's use the embed snippets from the code. I know it's not documented (my fault), but please take a look at other examples. It basically needs a marker in the Go file (ideally an example file so no testing code lands into the snippet):
```go
func A() {
  // uniqueMarker {
  if true { return nil }
  // }
}
```

And then in the markdown file use the `codeinclude` plugin, calling that unique marker (please see other examples).

> Here we created network for our container and added kafka to it, so they can communicate. Then we marked port 9092 for our internal usage.

Reviewer: ```suggestion
In the above code, we created a network for our container and attached kafka to it, so they can communicate. Then we marked port 9092 for our internal usage.
```

> First listener in slice will be written in `KAFKA_INTER_BROKER_LISTENER_NAME`  

Reviewer: ```suggestion
The first listener in the slice will be named as `KAFKA_INTER_BROKER_LISTENER_NAME`  
```

Author: Nope, it will be written in env parameter, not named as `KAFKA_INTER_BROKER_LISTENER_NAME`

> Every listener's name will be converted in upper case. Every name and port should be unique and will be checked in validation step.

Reviewer: It's not clear if the validation step is internal o must be added by the user. Could you clarify that?
```suggestion
Every listener's name will be converted in upper case. Every name and port should be unique and will be checked in a validation step.
```

> If you are not using this option or list is empty, there will be 2 default listeners with next addresses

Reviewer: ```suggestion
If you are not using this option or the listeners list is empty, there will be 2 default listeners with the following addresses and ports:
```
